File: strategy/java/Duck.java
/**
 * File Duck.java
 * 
 * @author kujaomega
 */

/**
 * This class will be the mother class for all ducks.
 * 
 * Class Duck
 */
public abstract class Duck {
    FlyBehavior flyBehavior;
    QuackBehavior quackBehavior;
    
    /**
     * This is the construct method for Duck.
     */
    public Duck(){
    }
    
    /**
     * This method will display the duck.
     */
    public abstract void display();
    
    /**
     * This method will make the duck fly.
     */
    public void performFly(){
        flyBehavior.fly();
    }
    
    /**
     * With this method, the duck will make the quack.
     */
    public void performQuack() {
        quackBehavior.quack();
    }
    
    /**
     * With this method, the duck will swim.
     */
    public void swim(){
        System.out.println("All ducks float, even decoys");
    }
    
    /**
     * This method will set the flybehavior.
     * 
     * @param FlyBehavior fb 
     */
    public void setFlyBehavior(FlyBehavior fb){
        flyBehavior=fb;
    }
    
    /**
     * This method will set the quack behavior.
     * 
     * @param qb 
     */
    public void setQuackBehavior(QuackBehavior qb){
        quackBehavior=qb;
    }
    
}

File: strategy/php/duck.php
<?php
/**
 * File duck.php
 *
 * @author kujaomega
 */

/**
 * This is the mother class.
 *
 * class Duck
 */
abstract class Duck
{
    /**
     * In this variable we save the flybehavior.
     *
     * @var flyBehavior $fly_behavior
     */
    public $fly_behavior;

    /**
     * In this variable we save the quackbehavior.
     *
     * @var quackbehavior $quack_behavior.
     */
    public $quack_behavior;

    /**
     * This is the construct of the class
     */
    public function __construct()
    {

    }

    /**
     * This is the class to display the ducks.
     */
    public abstract function display();

    /**
     * This class perform the quack.
     */
    public function performQuack()
    {
        $this->quack_behavior->quacks();
    }

    /**
     * This class perform the fly.
     */
    public function performFly()
    {
        $this->fly_behavior->fly();
    }

    /**
     * This class perform the swim.
     */
    public function swim()
    {
        echo( 'All ducks can swim' );
    }

    /**
     * With this method we can change any time we want the fly behavior.
     *
     * @param FlyBehavior $flybehavior This is the flybehavior to change for
     */
    public function setFlyBehavior( FlyBehavior $fly_behavior )
    {
        $this->fly_behavior = $fly_behavior;
    }

    /**
     * With this method we change the quack behavior.
     *
     * @param QuackBehavior $quack_behavior This is the quackbehavior to change
     */
    public function setQuackBehavior( QuackBehavior $quack_behavior )
    {
        $this->quack_behavior = $quack_behavior;
    }

}